util/src/journaldb/overlayrecentdb.rs

```diff
@@ -430,8 +434,51 @@ impl HashDB for OverlayRecentDB {
 		v.or_else(|| self.payload(key))
 	}
 
+	fn get_exec<'this>(
+		&'this self,
+		key: &'this H256,
+		f: &'this mut for<'a: 'this> FnMut(&'a [u8]),
+	) {
+		use ::std::cell::Cell;
+
+		let k = self.transaction_overlay.raw_ref(key);
+
+		if let Some((d, rc)) = k {
+			if rc > 0 {
+				f(d.as_ref());
+				return;
+			}
+		}
+
+		let was_called = Cell::new(false);
+
+		// This type annotation is necessary, the inferred lifetime is too restrictive otherwise.
+		// Possible bug in the Rust compiler.
+		let mut wrapper = |d: &[u8]| {
+			was_called.set(true);
+			f(d);
```

Comment:
borrows of a variable can last the entire lifetime of that variable, but my understanding the lifetime of the inner `ReadGuard` value has to be less than than `'this`, because it is declared within the function body.